src/MidiConversions.elm

```diff
@@ -74,30 +52,27 @@ accidentalToInt accidental =
             0
 
 
-flatten : List (List a) -> List a
-flatten list =
-    List.foldr (++) [] list
-
-
-toMidi : Array Pitch -> List Midi.Byte
-toMidi row =
-    row
+toMidi : Array Pitch -> List Midi.Types.Byte
+toMidi pitches =
+    pitches
         |> Array.toList
         |> List.map toMidiNumber
         |> List.map
             (\midiNumber ->
-                [ ( 0, Midi.NoteOn 0 midiNumber 64 )
-                , ( 2, Midi.NoteOff 0 midiNumber 0 )
+                [ ( 0, Midi.Types.NoteOn 0 midiNumber 64 )
+                , ( 2, Midi.Types.NoteOff 0 midiNumber 0 )
                 ]
             )
         |> flatten
-        |> Midi.SingleTrack 4
+        |> Midi.Types.SingleTrack 4
         |> recording
 
 
-toBase64EncodedMidi : Array Pitch -> MidiEncoding
-toBase64EncodedMidi bytes =
-    { contentType = "audio/midi"
-    , contentTransferEncoding = "base64"
-    , midiData = (encode (toMidi bytes))
-    }
+concatParts : String -> String -> String -> String
+concatParts contentType contentTransferEncoding midiData =
+    "data:" ++ contentType ++ ";" ++ contentTransferEncoding ++ "," ++ midiData
+
+
+createDataLink : Array Pitch -> String
+createDataLink =
+    toMidi >> BinaryBase64.encode >> concatParts "audio/midi" "base64"
```

Comment:
I like this style a bit better than using '|>'. Is this called "point free"? It looks like function composition.

Reply:
Exactly, it's reversed function composition.